src/Features/LanguageServer/Protocol/Handler/Rename/RenameHandler.cs

```diff
@@ -2,31 +2,31 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Composition;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Editor;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
-using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
 
-namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
-    internal class RenameHandler : ILspRequestHandler<RenameParams, WorkspaceEdit, Solution>
+    [ExportLspMethod(LSP.Methods.TextDocumentRenameName), Shared]
+    internal class RenameHandler : IRequestHandler<LSP.RenameParams, WorkspaceEdit>
     {
         private readonly IThreadingContext _threadingContext;
 
+        [ImportingConstructor]
         public RenameHandler(IThreadingContext threadingContext)
         {
             _threadingContext = threadingContext;
```

Comment:
threading context should no longer be needed

Reply:
Additionally you can remove the switch to main thread and comment before calling GetRenameInfoAsync as you no longer need to be on the UI thread